dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+# coding: utf-8
+"""Dragonfly Room Polygon."""
+from ._base import _Base
+from .properties import RoomPolygonProperties
+from .glazingparameter import GlazingParameter
+
+import honeybee.boundarycondition.boundary_conditions as bcs
+from honeybee.boundarycondition import _BoundaryCondition, Outdoors, Surface
+from honeybee.typing import tuple_with_length, float_positive
+from honeybee.room import Room
+from honeybee.aperture import Aperture
+
+from ladybug_geometry.geometry3d.face import Face3D
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+
+import math
+
+
+class RoomPolygon(_Base):
+    """A volume defined by an extruded floor plate, representing a single room or space.
+
+    Properties:
+        * name
+        * display_name
+        * geometry
+        * floor_to_ceiling_height
+        * boundary_conditions
+        * glazing_ratios
+        * glazing_parameters
+        * parent
+        * has_parent
+        * floor_segments
+        * segment_count
+        * floor_height
+        * ceiling_height
+        * volume
+        * floor_area
+        * exterior_wall_area
+        * exterior_aperture_area
+    """
+    __slots__ = ('_geometry', '_segment_count', '_floor_to_ceiling_height',
+                 '_boundary_conditions', '_glazing_ratios', '_glazing_parameters',
+                 '_parent')
+
+    def __init__(self, name, geometry, floor_to_ceiling_height, boundary_conditions=None,
+                 glazing_ratios=None, glazing_parameters=None, tolerance=None):
+        """A volume defined by an extruded floor plate, representing a single room.
+
+        Args:
+            name: Room name. Must be < 100 characters.
+            geometry: A single Face3D object representing the floor plate of the Room.
+            floor_to_ceiling_height: A number for the height above the floor where the
```

Comment:
I think it will be simpler we just use `height` as the argument name. You can add the details to docstring.

Reply:
That is going to create confusion with the `Story.floor_to_floor height`, then. These two types of heights are purposefully separate in the case that someone does not want to include the ceiling plenum space within the zone volume. For example, this is definitely what you want to do with a hospital, since the ventilation rate is based on the volume between floor and ceiling but there are often huge ceiling plenum spaces to accommodate all of the ventilation equipment. Other use-cases might be a 2-story building with a certain section that has a double-height gymnasium or auditorium (like a school). Having separate parameters for zone-level floor-to-ceiling height and Story floor-to-floor height will enable you to model a building like this with the Dragonfly object and not have to resort to Honeybee objects.